### PR TITLE
Add the dependency-zip workflow

### PR DESCRIPTION
**Pull request by estevao90**
It adds a new workflow to allow us to download a dependency zip from S3 if available.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,57 @@
...
+
+### [Dependency zip](.github/workflows/dependency-zip.yml)
+
+It allows us to download a dependency zip from S3 when available.
+
+#### Usage
+
+```yaml
+dependency-zip:
+    uses: stellarwp/github-actions/.github/workflows/dependency-zip.yml@dependency-zip
```
Comment (d4mation) [inline README.md]: We'll need to change the branch here in the docs.

**Review by d4mation — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (estevao90) [reply README.md]: Good catch. Fixed.

**Review by borkweb — APPROVED** [review]

File: .github/workflows/dependency-zip.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+          if [ $head_branch_exists -eq 1 ]; then
+            echo "dependency_branch=${{ needs.prepare.outputs.head-branch }}" >> $GITHUB_OUTPUT
+          else
+            base_branch_exists=$(git ls-remote --heads origin ${{ needs.prepare.outputs.base-branch }} | wc -l)
+
+            if [ $base_branch_exists -eq 1 ]; then
+              echo "dependency_branch=${{ needs.prepare.outputs.base-branch }}" >> $GITHUB_OUTPUT
+            else
+              echo "dependency_branch=main" >> $GITHUB_OUTPUT
```
Comment (borkweb) [inline .github/workflows/dependency-zip.yml]: Let's change main to a variable that can be passed in with main as the default. Some repos still have a base of master

**Review by borkweb — CHANGES_REQUESTED** [review]

File: .github/workflows/dependency-zip.yml (hunk 2 above)
Comment (estevao90) [reply .github/workflows/dependency-zip.yml]: Done!

**Review by borkweb — APPROVED** [review]
Excellent addition!!!

**Pull request merged**